dts/bindings/clock/nordic,nrf-hsfll.yaml

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Nordic nRF HSFLL
+
+  The HSFLL mixed-mode IP generates several clock frequencies in the range from
+  64 MHz to 400 MHz. HSFLL clock is fed with FLL16M (16 MHz). When working in
+  closed loop mode, its output frequency is calculated as:
+
+    f_hsfll = 16 MHz * multiplier
+
+compatible: "nordic,nrf-hsfll"
+
+include: [base.yaml, clock-controller.yaml]
+
+properties:
+  reg:
+    required: true
+
+  "#clock-cells":
+    const: 0
+
+  multiplier:
+    type: int
+    enum:
+      - 4
+      - 5
+      - 6
+      - 7
+      - 8
+      - 9
+      - 10
+      - 11
+      - 12
+      - 13
+      - 14
+      - 15
+      - 16
+      - 17
+      - 18
+      - 19
+      - 20
+      - 21
+      - 22
+      - 23
+      - 24
+      - 25
+    default: 6
+    description: |
+      Clock multiplication factor. Defaults to 6, reset default.
+
+  trim:
+    type: boolean
+    description: |
+      Enable trimming of the HSFLL VSUP, COARSE and FINE parameters.
```

Comment:
this is really a software config, wondering if Kconfig should be kept for it instead of using a DT property, thoughts?

Reply:
How about dropping this property and renaming the two others to `trims` and `trim-names`, and performing the trimming based on the presence of the `trims` property?

Reply:
> How about dropping this property and renaming the two others to `trims` and `trim-names`

That won't work, right? The property must be named `trims`.

Reply:
you can make use of `specifier-space` bindings property, but it looks like `names` doesn't work then...

Reply:
The problem I see with using the presence of ficrs/trims is that these are defined in soc dts files for convenience (so always there).